https://sources.debian.org/patches/chromium/120.0.6099.224-1~deb11u1/fixes/rust-clanglib.patch/
author: Andres Salomon <dilinger@debian.org>

Rust adds some new clang dependencies; specifically:

ninja: error: '../../third_party/llvm-build/Release+Asserts/lib/clang/17/lib/x86_64-unknown-linux-gnu/libclang_rt.builtins.a', needed by 'obj/third_party/protobuf/libprotoc_lib.a', missing and no known rule to make it

This is in the libclang-rt-14-dev package, but with a different (and
architecture-specific) path. So we special-case linux instead of
doing the same thing that upstream does w/ chromeos.


--- chromium-128.0.6613.137.orig/build/config/clang/BUILD.gn
+++ chromium-128.0.6613.137/build/config/clang/BUILD.gn
@@ -189,7 +189,22 @@ template("clang_lib") {
         }
       } else if (is_apple) {
         _dir = "darwin"
-      } else if (is_linux || is_chromeos) {
+      } else if (is_linux) {
+        _dir = "linux"
+        if (current_cpu == "x64") {
+          _suffix = "-x86_64"
+        } else if (current_cpu == "x86") {
+          _suffix = "-i386"
+        } else if (current_cpu == "arm64") {
+          _suffix = "-aarch64"
+        } else if (current_cpu == "arm") {
+          _suffix = "-armhf"
+        } else if (current_cpu == "ppc64") {
+          _suffix = "-powerpc64le"
+        } else {
+          assert(false)  # Unhandled cpu type
+        }
+      } else if (is_chromeos) {
         if (current_cpu == "x64") {
           _dir = "x86_64-unknown-linux-gnu"
         } else if (current_cpu == "x86") {

